pkg/types/aws/validation/platform.go

```diff
@@ -39,6 +39,10 @@ func ValidatePlatform(p *aws.Platform, fldPath *field.Path) field.ErrorList {
 		}
 	}
 
+	if p.HostedZoneRole != "" && p.HostedZone == "" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("hostedZoneRole"), p.HostedZoneRole, "may not specify a role to assume for hosted zone operations without also specifying a hosted zone"))
+	}
```

Comment:
In theory, we could validate the role specifically, but I would do the validation through the AWS API rather than with a regex because regexes like this [are error prone](https://github.com/openshift/api/pull/1480) (and unnecessary when we can have aws validate it for us).

On the other hand, I believe  abd6e17d49aaf37c0ac49016182a34856419755b essentially covers this validation. When the ARN role is bad, a pretty clear error message is returned to the user.